libs/api/domains/work-machines/src/lib/api-domains-work-machines.service.ts

```diff
@@ -0,0 +1,99 @@
+import { WorkMachinesClientService } from '@island.is/clients/work-machines'
+import { User } from '@island.is/auth-nest-tools'
+import { WorkMachine, WorkMachineCollection } from './models/getWorkMachines'
+import { Injectable } from '@nestjs/common'
+import { Action, ExternalLink } from './api-domains-work-machines.types'
+import { GetWorkMachineCollectionInput } from './dto/getWorkMachineCollection.input'
+import { GetWorkMachineInput } from './dto/getWorkMachine.input'
+import { GetDocumentsInput } from './dto/getDocuments.input'
+@Injectable()
+export class WorkMachinesService {
+  constructor(private readonly machineService: WorkMachinesClientService) {}
+
+  private mapRelToAction = (rel?: string) => {
+    switch (rel) {
+      case 'requestInspection':
+        return Action.REQUEST_INSPECTION
+      case 'changeStatus':
+        return Action.CHANGE_STATUS
+      case 'ownerChange':
+        return Action.OWNER_CHANGE
+      case 'supervisorChange':
+        return Action.SUPERVISOR_CHANGE
+      case 'registerForTraffic':
+        return Action.REGISTER_FOR_TRAFFIC
+      default:
+        return null
+    }
+  }
+
+  private mapRelToCollectionLink = (rel?: string) => {
+    switch (rel) {
+      case 'self':
+        return ExternalLink.SELF
+      case 'nextPage':
+        return ExternalLink.NEXT_PAGE
+      case 'excel':
+        return ExternalLink.EXCEL
+      case 'csv':
+        return ExternalLink.CSV
+      default:
+        return null
+    }
+  }
+
+  async getWorkMachines(
+    user: User,
+    input: GetWorkMachineCollectionInput,
+  ): Promise<WorkMachineCollection> {
+    const data = await this.machineService.getWorkMachines(user, input)
+
+    const value = data.value?.map(
+      (v) =>
+        ({
+          ...v,
+          ownerName: v.owner,
+          supervisorName: v.supervisor,
+        } as WorkMachine),
+    ) as Array<WorkMachine>
+
+    const links = data.links?.length
+      ? data.links.map((l) => {
+          return {
+            ...l,
+            rel: this.mapRelToCollectionLink(l.rel ?? ''),
+          }
+        })
+      : null
```

Comment:
Might want to re-model this kind of link stuff, since usually it is REST stuff which doesn't make much sense in the GraphQL world. 

Do you need it for anything other then downloads? Can you model downloadable stuff differently?

Reply:
Can you elaborate a bit more on what you mean?
All the buttons on the work machine detail page also use external links